### PR TITLE
Functional refactor

### PR DESCRIPTION
**Pull request by DavidBarker-EE**

### CODE REVIEW

**Comment by Andrew-Folga** [comment]
LGTM - I will wait for the removal of WIP flag before approving

File: .idea/codeStyles/Project.xml
Comment (Andrew-Folga) [inline .idea/codeStyles/Project.xml:1]: Do we need IDE specific files in repo? I am mindful of not committing vscode settings etc. WDYT?

File: .idea/codeStyles/codeStyleConfig.xml
Comment (Andrew-Folga) [inline .idea/codeStyles/codeStyleConfig.xml:1]: Same as above.

File: .idea/codeStyles/Project.xml
Comment (Andrew-Folga) [reply .idea/codeStyles/Project.xml:1]: Maybe we could add this to gitignore:

```
# VSCode settings
.vscode/

# IntelliJ IDEA settings
.idea/
*.iml
```

Comment (alanplatt) [reply .idea/codeStyles/Project.xml:1]: +1

Comment (swdpcomputing) [reply .idea/codeStyles/Project.xml:1]: +1

Comment (Andrew-Folga) [reply .idea/codeStyles/Project.xml:1]: Actually I think we have it added as exception:
`
!.idea/codeStyles
!.idea/runConfigurations
`

I am fine if everyone else is fine with it if it is only code styles etc.

**Review by swdpcomputing — DISMISSED** [review]

File: .idea/codeStyles/Project.xml
Comment (DavidBarker-EE) [reply .idea/codeStyles/Project.xml:1]: It is only codestyles, it makes sense that we version control these for anyone picking the project up it autoconfigures their IDE of choice.

**Review by Andrew-Folga — DISMISSED** [review]
LGTM

**Review by swdpcomputing — DISMISSED** [review]

**Review by swdpcomputing — APPROVED** [review]

**Pull request merged**